backend/lib/endpoints/posts.js

```diff
@@ -176,13 +176,11 @@ async function routes(app) {
           ? [
             { $match: { $and: filters, $text: { $search: keywords } } },
             { $sort: { score: { $meta: "textScore" } } },
-          ]
-          : sort
-            ? [
-              { $sort: { [sort]: order === "asc" ? 1 : -1 } }
-            ]
-            : [{ $match: { $and: filters } }, { $sort: { _id: -1 } }];
+          ] : [{ $match: { $and: filters } }, { $sort: { _id: -1 } }];
       /* eslint-enable sort-keys */
+      if (sort) {
```

Comment:
Hi @linadinh2602 , I just realized this is complex than I thought
> when the location is passed in, do we sort based on distance or the sort argument (likes, views, etc.)?
> when the keyword is passed in, do we sort based on keyword relevant or sort argument (likes, views, etc.)?

The current change appends a new `sort` clause after the sortAndFilterSteps, which could result in 2 `sort` steps for mongoDB to process I guess, like showing below, in which case the 1st sort will be overwritten by the 2nd one I assume, we may need to understand the requirement and combine them

![image](https://user-images.githubusercontent.com/44758313/109561240-c04c3b00-7a91-11eb-956f-bcdf96b861a6.png)

Reply:
Would it make more sense in each sort case to check if sort exists and if it does sort by sort param, otherwise sort by id?

`[{ $match: { $and: filters } }, sort ?{ $sort: { [sort]: order === "asc" ? 1 : -1 } } : { $sort: { _id: -1 } }];`

Reply:
we have following sort mechanisms right now in the endpoint
```
by distance if location is present
by relevance if keyword is present
by 'likes/views/shares/..." from the drop-down menu
```
I think the PRD needs to define these behavior and based on the definition we can implement

Reply:
e.g. if we combine location with the drop-down like below, since no `$maxDistance` is indicated I assume the `$geoNear` will return all records, if `sort === 'likes'` is present at the same time, will the result be any difference from the result sorted by 'likes' itself?
```      const sortAndFilterSteps = location
        ? [
            {
              $geoNear: {
                distanceField: "distance",
                key: "author.location.coordinates",
                near: {
                  $geometry: {
                    coordinates: location.coordinates,
                    type: "Point",
                  },
                },
                query: { $and: filters },
              },
            },
            sort
              ? { $sort: { [sort]: order === "asc" ? 1 : -1 } }
              : { $sort: { distance: 1, _id: -1 } },
          ]
```

Reply:
We will be resolving this on the front-end by making sorting mutually exclusive